.github/workflows/test.yml

```diff
@@ -6,4 +6,19 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - run: echo "TODO Run automated tests"
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+      - name: Set up JDK 14
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Run maven-jar tests
+        run: mvn -B test --file maven-jar/pom.xml
+      - name: Run maven-service tests
```

Comment:
Same thoughts here, as individual projects will probably only be "one of" the list of project types so maybe we start introducing subduers for each project type...

```
.github/workflows/maven-jar/*.yml
.github/workflows/maven-service/*.yml
.github/workflows/html-app/*.yml
```